From d1359cf89f1d214768c6380e0e82414c9e8c3d5c Mon Sep 17 00:00:00 2001
From: rkannapp <r.kannappan@intel.com>
Date: Mon, 18 Aug 2014 18:26:16 +0530
Subject: Charger: Improve screen on/off scenarios

BZ: 217720

In charger-mode, power off/on screen as required, for efficient charging.
Turn on screen on power-key press rather than on key-release.
On charge-cable removal, kick animation to show the correct battery level

Change-Id: I3e030ad7fdf89299fd6aec5698a4db3f448063e9
Category: aosp improvement
Domain: AOSP-Framework-CoreService
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Pavan Kumar S <pavan.kumar.s@intel.com>
Signed-off-by: Jenny TC <jenny.tc@intel.com>
Signed-off-by: rkannapp <r.kannappan@intel.com>
---
 healthd/healthd_mode_charger.cpp |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/healthd/healthd_mode_charger.cpp b/healthd/healthd_mode_charger.cpp
index d7fda7a..4791516 100644
--- a/healthd/healthd_mode_charger.cpp
+++ b/healthd/healthd_mode_charger.cpp
@@ -539,10 +539,14 @@ static void process_key(struct charger *charger, int code, int64_t now)
                  */
                 set_next_key_check(charger, key, POWER_ON_KEY_TIME);
             }
+            /* Turn on display and kick animation on power-key press
+             * rather than on key release
+             */
+             kick_animation(charger->batt_anim);
+             request_suspend(false);
         } else {
             /* if the power key got released, force screen state cycle */
             if (key->pending) {
-                request_suspend(false);
                 kick_animation(charger->batt_anim);
             }
         }
@@ -565,6 +569,10 @@ static void handle_power_supply_state(struct charger *charger, int64_t now)
         return;
 
     if (!charger->charger_connected) {
+        /* Last cycle would have stopped at the extreme top of battery-icon
+         * Need to show the correct level corresponding to capacity.
+         */
+        kick_animation(charger->batt_anim);
         request_suspend(false);
         if (charger->next_pwr_check == -1) {
             charger->next_pwr_check = now + UNPLUGGED_SHUTDOWN_TIME;
-- 
1.7.9.5

